samples/grpc/local-drone-control-scala/native-image/README.md

```diff
@@ -0,0 +1,24 @@
+# GraalVM Native Image
+
+The Local Drone Control service is configured for building a [GraalVM Native Image], for low
+resource usage, faster starts, and smaller deployments.
+
+[GraalVM Native Image]: https://www.graalvm.org/latest/reference-manual/native-image/
+
+
+## Native build
+
+To create a native image for the current build platform and architecture:
+
+```
+sbt nativeImage
+```
```

Comment:
Getting 

> The program tried to reflectively invoke method public local.drones.DeliveriesQueue$DeliveryInProgress(java.lang.String,java.lang.String,java.time.Instant) without it being registered for runtime reflection. Add it to the reflection metadata to solve this problem.

So something still missing from the reflect json?

Reply:
Yeah, missing. This is something we'll want to automatically generate the config ourselves. With the tracing agent, need to interact in all possible ways. In this case, it needs an existing queue with both waiting deliveries and deliveries in progress, so that the constructors are used when deserializing — missed exercising this.

I'll look through the state and event classes and see if any others may be missing config for constructors.

Reply:
Updated and tested. Should be good now. I've also tested a couple of error paths. Good to try it out fully with what you've been testing already.

Reply:
Now I'm, instead getting an error, same on both mac and linux:
>JVM graalvm:17.0.8 not found in index: No graalvm version matching '17.0.8' found

Reply:
I saw the Java version logging 17.0.8+9.1 at some point so tried that, but it didn't help

Reply:
Hmm, sbt-native-image uses coursier to download graalvm. Maybe just an outdated index? But not sure how running it earlier would have worked then.

The coursier cache should be in `~/Library/Caches/Coursier` for mac, `~/.cache/coursier` for linux. Could have a look to see what's in there.

Checking index using the downloaded coursier, from `local-drone-control-scala`:

```
target/native-image-internal/coursier java --jvm-index cs --available | grep 'graalvm'
```

And if it's just updating the index needed, then either delete it:

```
rm ~/Library/Caches/Coursier/v1/https/github.com/coursier/jvm-index/raw/master/index.json
```

Or looks like can set TTL to 0:

```
target/native-image-internal/coursier java --jvm-index cs --available --ttl 0 | grep 'graalvm'
```

Reply:
Huh, on my office workstation it works fine. So something with resolution just yesterday when I tried from machines at home?

Checking the index on the linux machine, I see graalvm:17.0.8 listed, still failing, but removing the index sorted the problem and now it works. Thanks!